### PR TITLE
Data Lasso in React

### PR DESCRIPTION
**Pull request by tgilan**
# Data Lasso in React

## Why
Because Data Lasso is better aligned to be used with React, now that is follows Flux convention.

## How
- Almost everything that was a Backbone View is now a React component. Wins are in the structure of more complicated UI bits, such as Uploader. Writing it in React (as opposed to Backbone view) allowed me to break it down into a set of smaller components with more granular functionality each.
- Certain things were Backbone Views, but did not leverage it at all, such as Graph View. The whole purpose of it is initializing and settings up THREE.js - so it being a Class, not a view or a component seemed more appropriate.
- Templates are gone. All markup is now in JSX.
- Build process uses `reactify` - browserify transform for React. I did not spend much time on optimizing that step, as I see that Data Lasso will benefit from using WebPack in the future.

## Future action items
- Get rid of Backbone entirely (it is still used for store and events in mouse helper)
- Now that Graph is not a component or a view, all THREE.js related code can be abstracted away into it's own space. I did not do it in this PR to keep the diff lighter.

## Testing
- It should just work as usual

@heylookltsme @kmck

### CODE REVIEW

File: src/components/Uploader.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        )
+    }
+});
+
+/**
+ * ## Toggles React component
+ *
+ * Renders a set of toggles that are used in the Uploader
+ */
+const Toggles = React.createClass({
```
Comment (heylookltsme) [inline src/components/Uploader.js]: Ok to have multiple classes per file? Seems fine for data lasso, but I know that's generally a no-no for us. Is it a common convention in react apps?

**Comment by heylookltsme** [comment]
Works great! I think that Uploader class might benefit from being broken up into smaller files, but it's no biggie. 

👍

File: src/components/Uploader.js (hunk 1 above)
Comment (tgilan) [inline src/components/Uploader.js]: That's a good point.

Airbnb React/JSX Style Guide, which seems like a reasonable guide, suggests that all components/classes be in their own files, with exception of stateless functional components.

In light of that (and other style guide instructions), I refactored file structure
 - Each component is in it's file (stateless components excluding)
 - Directories in components correspond to logical separation of concerns in the UI
 - Everything that has to do with 3 dimensional graph was moved to `graph` directory - there was no sense to have that code mixed up with UI code.
 - Got rid of `lib/Class` - a polyfill that is obsolete now that I use es6

Thanks for pointing that out!

File: src/graph/helpers/selection.js [hunk 2]
```diff
@@ -37,9 +36,8 @@ var planeSize = 2000;
  */
 var lassoPoints = 4;
 
-var SelectionHelper = Class.extend({
-
-    initialize: function (scene, camera, mouse) {
+var SelectionHelper = class {
+    constructor (scene, camera, mouse) {
```
Comment (kmck) [inline src/graph/helpers/selection.js]: nitpick, but maybe `constructor(scene, camera, mouse)` etc...

**Comment by kmck** [comment]
looks awesome! 👍

**Pull request merged**